### PR TITLE
Add tooling to improve docs

### PR DESCRIPTION
**Pull request by s-weigand**
Here is the requested docsting PR 😉 
- Added [`interrogate`](https://github.com/econchick/interrogate) (coverage for docstrings) to `pre-commit`
- Added [`darglint`](https://github.com/terrencepreilly/darglint) (helps keeping function signature and docstrings in sync) to `pre-commit` (currently commented out so the CI will pass)
- Added docstring instructions to contributing guide

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -73,6 +73,10 @@
 # source_suffix = ['.rst', '.md']
 source_suffix = ".rst"
 
+linkcheck_ignore = [
+    r"https://github\.com/.+?#.+",
```
Comment (jsnel) [inline docs/source/conf.py:77]: Why ignore github.com links?

**Review by jsnel — COMMENTED** [review]
Looks good, but would appreciate an up-front estimate of the work to update documentation if this is to be merged.

File: docs/source/conf.py (hunk 1 above)
Comment (s-weigand) [reply docs/source/conf.py:77]: They are generated by frontend rendering, so they can't be found by the link checker.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -33,6 +34,16 @@ repos:
   #       # this is needed due to the following issue:
   #       # https://github.com/PyCQA/pydocstyle/issues/368
   #       args: [--ignore-decorators=wrap_func_as_method]
+  # - repo: https://github.com/terrencepreilly/darglint
+  #   rev: v1.5.5
+  #   hooks:
+  #     - id: darglint
+  - repo: https://github.com/econchick/interrogate
```
[the remark this answers is not in the document]
Comment (s-weigand) [reply .pre-commit-config.yaml:41]: This just prevents adding new things w/o docstrings.
`darglint` and `pydocstyle`, will be a lot of work, which is why they are commented out for the moment.

**Pull request merged**